transport/grpc/handler.go

```diff
@@ -0,0 +1,154 @@
+package grpc
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/yarpc/yarpc-go/transport"
+	"github.com/yarpc/yarpc-go/transport/internal"
+
+	"strings"
+
+	"fmt"
+
+	"io"
+
+	"errors"
+
+	"net/url"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+	gtransport "google.golang.org/grpc/transport"
+)
+
+var grpcOptions transport.Options
+
+type handler struct {
+	Handler transport.Handler
+	Deps    transport.Deps
+}
+
+// Handle the grpc request and convert it into a YARPC request
+// dec ('decode') will pass through the request body in raw bytes using the PassThroughCodec
+func (h handler) Handle(
+	srv interface{},
+	ctx context.Context,
+	dec func(interface{}) error,
+	interceptor grpc.UnaryServerInterceptor,
+) (interface{}, error) {
+	treq, extractErr := getTRequest(ctx, dec)
+	if extractErr != nil {
+		return nil, extractErr
+	}
+
+	// TODO handle deadlines
+	// TODO handle validation
+
+	start := time.Now()
+
+	return callHandler(h, start, ctx, treq)
+}
+
+func getTRequest(ctx context.Context, msgBodyDecoder func(interface{}) error) (*transport.Request, error) {
+	service, procedure, err := getServiceAndProcedure(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	requestBody, err := getMsgBody(msgBodyDecoder)
+	if err != nil {
+		return nil, err
+	}
+
+	treq := &transport.Request{
+		Service:   service,
+		Procedure: procedure,
+		Caller:    "yarpc",
+		Encoding:  transport.Encoding("raw"),
+		Body:      requestBody,
+	}
+
+	return treq, nil
+}
+
+func getServiceAndProcedure(ctx context.Context) (service, procedure string, err error) {
+	stream, ok := gtransport.StreamFromContext(ctx)
+	if !ok {
+		return "", "", errors.New("Could not extract stream information from context")
+	}
+
+	streamMethod := stream.Method()
+
+	return getServiceAndProcedureFromMethod(streamMethod)
+}
+
+func getServiceAndProcedureFromMethod(streamMethod string) (service, procedure string, err error) {
+	if streamMethod != "" && streamMethod[0] == '/' {
```

Comment:
Handle `== ""` separately (presumably with failure) and then strip the `/` separately.